addons/binding/org.openhab.binding.wso2iots/src/main/java/org/openhab/binding/wso2iots/handler/wso2iotsHandler.java

```diff
@@ -0,0 +1,390 @@
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            Runnable runnable = new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        // Request new data
+                        iotsResponseTemp = updateBuildingMonitorData(wso2iotsBindingConstants.TEMPERATURE);
+                        iotsResponseLight = updateBuildingMonitorData(wso2iotsBindingConstants.LIGHT);
+                        iotsResponseMotion = updateBuildingMonitorData(wso2iotsBindingConstants.MOTION);
+                        iotsResponseHumidity = updateBuildingMonitorData(wso2iotsBindingConstants.HUMIDITY);
+
+                        // Update all channels from the updated data
+                        updateChannel(wso2iotsBindingConstants.TEMPERATURE, iotsResponseTemp);
+                        updateChannel(wso2iotsBindingConstants.LIGHT, iotsResponseLight);
+                        updateChannel(wso2iotsBindingConstants.MOTION, iotsResponseMotion);
+                        updateChannel(wso2iotsBindingConstants.HUMIDITY, iotsResponseHumidity);
+
+                        // for (Channel channel : getThing().getChannels()) {
+                        // updateChannel(channel.getUID().getId(), iotsResponse);
+                        // }
+                    } catch (Exception e) {
+                        logger.error("Exception occurred during execution: {}", e.getMessage(), e);
+                    }
+                }
+            };
+
+            wso2iotsConfiguration config = getConfigAs(wso2iotsConfiguration.class);
+            int delay = (config.refresh != null) ? config.refresh.intValue() : DEFAULT_REFRESH_PERIOD;
+            refreshJob = scheduler.scheduleWithFixedDelay(runnable, 0, delay, TimeUnit.MINUTES);
+        }
+    }
+
+    /**
+     * Update the channel from the last building monitor data retrieved
+     *
+     * @param channelId the id identifying the channel to be updated
+     */
+    private void updateChannel(String channelId, BigDecimal iotsResponse) {
+        if (isLinked(channelId)) {
+            /*
+             * Object value;
+             * try {
+             * value = getValue(channelId, iotsResponse);
+             * } catch (Exception e) {
+             * logger.debug("Station doesn't provide {} measurement", channelId.toUpperCase());
+             * return;
+             * }
+             */
+
+            State state = null;
+            if (iotsResponse == null) {
+                state = UnDefType.UNDEF;
+            } else if (iotsResponse instanceof BigDecimal) {
+                state = new DecimalType(iotsResponse);
+
+            } else {
+                logger.warn("Update channel {}: Unsupported value type {}", channelId,
+                        iotsResponse.getClass().getSimpleName());
+            }
+            logger.debug("Update channel {} with state {} ({})", channelId, (state == null) ? "null" : state.toString(),
```

Comment:
it should actually work if you pass even null arguments to the logger, it will print null